### PR TITLE
🐛 Change `LightningLoggerBase` to `Logger`

### LINKED ISSUE #972: pytorch_lightning.loggers import LightningLoggerBase
In data_monitor.py
```
pytorch_lightning.loggers import LightningLoggerBase
```
I always get the error message cannot import name 'LightningLoggerBase' from 'pytorch_lightning.loggers'

but would be working if 
```
from pytorch_lightning.loggers.base import LightningLoggerBase
```

Or if you update latest pytorch_lightning, they removed LightningLoggerBase completely, and use something else simpler.

### PR DESCRIPTION
**Pull request by julien-blanchon**
## What does this PR do?

Fix `cannot import name 'LightningLoggerBase' from 'pytorch_lightning.loggers'`. Since `pytorch_lightning.loggers.base.LightningLoggerBase` is deprecated over `pytorch_lightning.loggers.base.Logger`.
Fixes #972 

## Before submitting

- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [ ] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by senarvi — APPROVED** [review]

**Comment by senarvi** [comment]
If we want to maintain compatibility to older versions, then we could do something like this:

```python
try:
    from pytorch_lightning.loggers import Logger
except ImportError:
    from pytorch_lightning.loggers import LightningLoggerBase
    Logger = LightningLoggerBase
```

**Comment by julien-blanchon** [comment]
@senarvi I did commit your idea to maintain older version compatibility

**Comment by D0miH** [comment]
Any updates on this? Having to change `LightningLoggerBase` to `Logger` every time I re-install the package is kinda tedious.

**Comment by tedtroxell** [comment]
Are we able to get this merged? I'm willing to help out any way possible. It seems like not being able to use lightning bolts with the latest pytorch lightning seems like a big deal

File: pl_bolts/callbacks/data_monitor.py [hunk 1]
```diff
@@ -14,6 +14,12 @@
 from pl_bolts.utils.stability import under_review
 from pl_bolts.utils.warnings import warn_missing_pkg
 
+try:
+    from pytorch_lightning.loggers import Logger
+except ImportError:
+    from pytorch_lightning.loggers import LightningLoggerBase
+    Logger = LightningLoggerBase
```
Comment (Borda) [inline pl_bolts/callbacks/data_monitor.py:21]:
```suggestion
    from pytorch_lightning.loggers import LightningLoggerBase as Logger
```

**Pull request merged**